docker/tools/run_services_tests.sh

```diff
@@ -27,6 +27,16 @@ check_pdo_runtime_env
 export no_proxy=$PDO_HOSTNAME,$PDO_LEDGER_ADDRESS,$no_proxy
 export NO_PROXY=$PDO_HOSTNAME,$PDO_LEDGER_ADDRESS,$NO_PROXY
 
+# -----------------------------------------------------------------
+yell copy sgx keys
+# -----------------------------------------------------------------
+# this collateral *must* be copied before configuring the services
+# as it will be included in the service/enclave toml files
+[ -z "$(ls -A ${XFER_DIR}/services/keys/sgx/)" ] ||\
+    cp ${XFER_DIR}/services/keys/sgx/* ${PDO_SGX_KEY_ROOT}
+# refresh the environment variables (necessary for SGX-related ones)
+source /project/pdo/tools/environment.sh
+
```

Comment:
why are we source'ing environment twice?

Reply:
The problem is that the SPID variables are not set properly before the collateral is available.

I'll go a step further and do the following:
- avoid double-sourcing the env
- remove all SPID env vars in PDO
- make sure that when the collateral is needed, it is taken from PDO_SGX_KEY_ROOT

This will clearly add complexity to the PR, and remove complexity from the dev environment.

Reply:
that would make life much better. i think thats what we talked about doing anyway.

Reply:
Looking at this more closely... my original concern still holds. there is no overall design (at least none that i can grok). we have some environment variables that are file contents, we have some that are paths and some that are file names. Of course the code is excessively complicated. First job... remove PDO_SPID and PDO_SPID_API_KEY completely; these should be files not environment variables. Second job, remove PDO_ENCLAVE_CODE_SIGN_PEM, it should be a standard named file in the PDO_SGX_KEY_ROOT directory (ALL of the sgx keys should have a standard name in that directory... completely consistent with the way we handle the four CCF keys/certificates).